sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs

```diff
@@ -383,6 +346,136 @@ protected EventProcessorClient()
+                    await InitializeProcessingForPartitionAsyncHandler(initializationContext).ConfigureAwait(false);
+
+                    startingPosition = initializationContext.DefaultStartingPosition;
+                }
+
+                var ownership = InstanceOwnership[context.PartitionId];
+
+                if (ownership.Offset.HasValue)
+                {
+                    startingPosition = EventPosition.FromOffset(ownership.Offset.Value);
+                }
+                else if (ownership.SequenceNumber.HasValue)
+                {
+                    startingPosition = EventPosition.FromSequenceNumber(ownership.SequenceNumber.Value);
+                }
+
+                // TODO: it might be troublesome to let the users add running tasks by themselves.  If the user adds a custom
+                // processing task that's not RunPartitionProcessingAsync, how would the base stop it?  It would not have a cancellation
+                // token to do so.
+
+                ActivePartitionProcessors[context.PartitionId] = RunPartitionProcessingAsync(context.PartitionId, startingPosition, Options.MaximumReceiveWaitTime, Options.RetryOptions, Options.TrackLastEnqueuedEventInformation);
+            }
+            catch (Exception)
+            {
+                // If processing task creation fails, we'll try again on the next time this method is called.  This should happen
+                // on the next load balancing loop as long as this instance still owns the partition.
+                // TODO: delegate the exception handling to an Exception Callback.  Do we really need a try-catch here?
+            }
+        }
+
+        /// <summary>
+        ///   The handler to be called once event processing stops for a given partition.
+        /// </summary>
+        ///
+        /// <param name="reason">The reason why the processing for the associated partition is being stopped.</param>
+        /// <param name="context">The context in which the associated partition was being processed.</param>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        protected async override Task ProcessingForPartitionStoppedAsync(ProcessingStoppedReason reason,
+                                                                         PartitionContext context)
+        {
+            if (ProcessingForPartitionStoppedAsyncHandler != null)
+            {
+                try
+                {
+                    var stopContext = new PartitionProcessingStoppedContext(context, reason);
+                    await ProcessingForPartitionStoppedAsyncHandler(stopContext);
+                }
+                catch (Exception)
+                {
+                    // TODO: delegate the exception handling to an Exception Callback.
+                    // Maybe we should just surface the exception.
```

Comment:
Remember, we don't signal for errors in user code (the handlers) but our own code should signal the error handler.

Reply:
That was the long and winding result of the service review, sayeth the architects in the room, particularly Jeff.

Reply:
Yeah. You'll see stuff like that here because I didn't try updating the error handling behavior (this PR already has a lot of stuff). I'll do that in another PR.